canopen/objectdictionary/__init__.py

```diff
@@ -108,6 +108,9 @@ def __getitem__(
         """Get object from object dictionary by name or index."""
         item = self.names.get(index) or self.indices.get(index)
         if item is None:
+            if isinstance(index, str) and '.' in index:
+                parts = index.split('.')
+                return self[parts[0]][".".join(parts[1:])]
```

Comment:
More compact and also faster variant:

```suggestion
                idx, sub = index.split('.', maxsplit=1)
                return self[idx][sub]
```

Reply:
Using positional-only arguments (`split('.', 1)`), is even faster, but I would prefer the keyword argument for readability.

Reply:
Thank you. I will update the PR